crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -426,73 +448,93 @@ where
             "tx_orchestrator_good_response",
-            "Total number of good responses Transaction Orchestrator generates, group by request type",
-            &["request_type"],
+            "Total number of good responses Transaction Orchestrator generates, group by tx type",
+            &["tx_type"],
             registry
         )
         .unwrap();
 
-        let good_response_wait_for_effects_cert =
-            good_response.with_label_values(&["wait_for_effects_cert"]);
-        let good_response_wait_for_local_execution =
-            good_response.with_label_values(&["wait_for_local_execution"]);
+        let good_response_single_writer =
+            good_response.with_label_values(&[TX_TYPE_SINGLE_WRITER_TX]);
+        let good_response_shared_object = good_response.with_label_values(&[TX_TYPE_SHARED_OBJ_TX]);
 
         let req_in_flight = register_int_gauge_vec_with_registry!(
             "tx_orchestrator_req_in_flight",
-            "Number of requests in flights Transaction Orchestrator processes, group by request type",
-            &["request_type"],
+            "Number of requests in flights Transaction Orchestrator processes, group by tx type",
+            &["tx_type"],
             registry
         )
         .unwrap();
 
-        let req_in_flight_wait_for_effects_cert =
-            req_in_flight.with_label_values(&["wait_for_effects_cert"]);
-        let req_in_flight_wait_for_local_execution =
-            req_in_flight.with_label_values(&["wait_for_local_execution"]);
+        let req_in_flight_single_writer =
+            req_in_flight.with_label_values(&[TX_TYPE_SINGLE_WRITER_TX]);
+        let req_in_flight_shared_object = req_in_flight.with_label_values(&[TX_TYPE_SHARED_OBJ_TX]);
+
+        let request_latency = HistogramVec::new_in_registry(
+            "tx_orchestrator_request_latency",
+            "Time spent in processing one Transaction Orchestrator request",
+            &["tx_type"],
+            registry,
+        );
+        let wait_for_finality_latency = HistogramVec::new_in_registry(
+            "tx_orchestrator_wait_for_finality_latency",
+            "Time spent in waiting for one Transaction Orchestrator request gets finalized",
+            &["tx_type"],
+            registry,
+        );
+        let local_execution_latency = HistogramVec::new_in_registry(
+            "tx_orchestrator_local_execution_latency",
+            "Time spent in waiting for one Transaction Orchestrator gets locally executed",
+            &["tx_type"],
+            registry,
+        );
 
         Self {
-            total_req_received_wait_for_effects_cert,
-            total_req_received_wait_for_local_execution,
-            good_response_wait_for_effects_cert,
-            good_response_wait_for_local_execution,
-            req_in_flight_wait_for_effects_cert,
-            req_in_flight_wait_for_local_execution,
+            total_req_received_single_writer,
```

Comment:
What is the reason for using 2 objects on the same metric with different labels? It seems all usages are wrapped by `if tx.contains_shared_object() { ... } else { ... }`, so I think we may as well have e.g. `total_req_received.with_label_values(&[get_tx_type_for_metric(&tx)])...`

Reply:
no specific reason. If you feel strong with using labels I can change

Reply:
resolved offline, comment added in https://github.com/MystenLabs/sui/pull/8038/files#diff-f12c25716656fb79726beee3513685fb66fa1dcb746df71cf0ed765e7464edd8R473-R475